ceph/ceph_admin/cephadm_ansible.py

```diff
@@ -101,6 +107,14 @@ def execute_playbook(self, playbook, extra_vars=None, extra_args=None):
         if extra_args:
             cmd += config_dict_to_string(extra_args)
 
+        # Check for Disconnected installation
+        for ceph in self.cluster.get_nodes():
+            if ceph.role == "network-node":
+                #Fetch network node hostname
+                network_node = self.cluster.get_ceph_object("network-node").node
+                network_node_hostname=network_node.hostname
+                cmd += f" -e 'custom_repo_url=http://{network_node_hostname}/repo/ceph-Tools'"
```

Comment:
If I understand correctly, in the case of disconnected install the following is true
- ceph tools repo is internally hosted
- container image registry is internally hosted

If correct, then we don't need to add any code changes and the testing can be done via the CLI options

`--add-repo` can be used to the tools repo

In `.cephci.yaml`, you would need to modify the `registry_credentials` section to point to the right one.